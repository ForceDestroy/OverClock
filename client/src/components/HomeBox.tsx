import { Anchor, Box, Button, Image, Text } from 'grommet';
import UIResources from '../resources/UIResources';

function HomeBox() {
  return (
    <Box
      pad="medium"
      gap="medium"
      width="100vw"
      height="100vh"
      alignSelf="center"
      align="center"
      animation={{ type: 'zoomIn', duration: 1000 }}
    >
      <Box align="center" alignContent="center" height="15vh">
        <Image
          src="/src/assets/images/logo.png"
          fit="cover"
          fill
          alt="overclock_logo"
        />
      </Box>
      <Text size="3xl">Workplace Management Accelerated</Text>
      <Anchor
        icon={
          <Button
            label={
              <Box width="20em">
                <br />
                <Text color="white" weight="bold" size="large">
                  Get Started
                </Text>
                <br />
              </Box>
            }
            color={UIResources.PALETTE.GRADIENT}
            primary
            size="medium"
            justify="center"
            alignSelf="center"
            margin="small"
            data-testid="getStartedButton"
          />
        }
        href="/Login"
      />
    </Box>
  );
}

export default HomeBox;
