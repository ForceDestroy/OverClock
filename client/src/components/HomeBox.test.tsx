import { render, screen } from '@testing-library/react';
import { describe, it } from 'vitest';

import HomeBox from './HomeBox';

describe('HomeBox', () => {
  it('Renders HomeBox to match snapshot', () => {
    // Arrange
    const snap = render(<HomeBox />);
    // Assert
    expect(snap).toMatchSnapshot();
  });
  it('Renders the navigation button', () => {
    // Arrange
    render(<HomeBox />);

    // Act
    // No Act for this Test

    // Assert
    const loginButton = screen.getByText('Get Started');
    expect(loginButton).not.toBeNull();
  });
});
