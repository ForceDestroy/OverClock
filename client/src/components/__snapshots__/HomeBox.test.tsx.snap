// Vitest Snapshot v1

exports[`HomeBox > Renders HomeBox to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="StyledBox-sc-13pk1d4-0 eFkftZ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 ewQYxc"
        >
          <img
            alt="overclock_logo"
            class="StyledImage-sc-ey4zx9-0 fiharU"
            src="/src/assets/images/logo.png"
          />
        </div>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
        />
        <span
          class="StyledText-sc-1sadyjn-0 ijIcEi"
        >
          Workplace Management Accelerated
        </span>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
        />
        <a
          class="StyledAnchor-sc-1rp7lwl-0 hklEFB"
          href="/Login"
        >
          <button
            class="StyledButton-sc-323bzc-0 jMQIxB"
            data-testid="getStartedButton"
            type="button"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 kObcQa"
            >
              <br />
              <span
                class="StyledText-sc-1sadyjn-0 cKTRLX"
              >
                Get Started
              </span>
              <br />
            </div>
          </button>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="StyledBox-sc-13pk1d4-0 eFkftZ"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 ewQYxc"
      >
        <img
          alt="overclock_logo"
          class="StyledImage-sc-ey4zx9-0 fiharU"
          src="/src/assets/images/logo.png"
        />
      </div>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
      />
      <span
        class="StyledText-sc-1sadyjn-0 ijIcEi"
      >
        Workplace Management Accelerated
      </span>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ceHRnz"
      />
      <a
        class="StyledAnchor-sc-1rp7lwl-0 hklEFB"
        href="/Login"
      >
        <button
          class="StyledButton-sc-323bzc-0 jMQIxB"
          data-testid="getStartedButton"
          type="button"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 kObcQa"
          >
            <br />
            <span
              class="StyledText-sc-1sadyjn-0 cKTRLX"
            >
              Get Started
            </span>
            <br />
          </div>
        </button>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
